### PR TITLE
Feature/refactor rule builder

### PR DESCRIPTION
**Pull request by assuntad23**
### Why you made the changes (e.g. references to GitHub issues being fixed):

Breaking out RuleCollectionBuilder into more modular components is the more reusable, readable, modern way to write this class.

### A description of the implementation of the changes.

Moved standalone components of RuleCollectionBuilder to their own classes.

### How to test the changes, if you haven't included specific tests already.

Test various functions in the RuleCollectionBuilder to make sure there are no unexpected errors. Following these tutorials may be useful: 
- https://training.galaxyproject.org/training-material/topics/galaxy-data-manipulation/tutorials/upload-rules/tutorial.html
- https://training.galaxyproject.org/training-material/topics/galaxy-data-manipulation/tutorials/collections/tutorial.html
- https://training.galaxyproject.org/training-material/topics/galaxy-data-manipulation/tutorials/processing-many-samples-at-once/tutorial.html

### CODE REVIEW

**Comment by bgruening** [comment]
Cool, thanks @assuntad23! What do you think about adding Vue tests to those components?

File: .ci/jenkins/api-py3 [hunk 1]
```diff
@@ -1 +0,0 @@
-api
```
Comment (nsoranzo) [inline .ci/jenkins/api-py3]: I'd suggest a rebase to get rid of the changes in the `.ci/` directory.

Comment (jmchilton) [reply .ci/jenkins/api-py3]: +1 - if you need help with the rebase process @assuntad23 let me know.

**Comment by jmchilton** [comment]
@assuntad23 can you work through the JS lint issues https://circleci.com/gh/galaxyproject/galaxy/82879?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-checks-link.

You can run `make client-lint` from the galaxy directory to get a list of the issues locally. It will be worth looking at the Makefile rule and learning to run the yarn commands directly also.

**Comment by jmchilton** [comment]
@bgruening Given these components all have integration test coverage I think writing unit tests for each (given they aren't being modified) would be a bit overkill at this time. But once we start modifying the components I'll work with @assuntad23 on making sure the modified components have Vue tests. Does that sound fair?

**Comment by jmchilton** [comment]
@galaxybot test this

**Pull request merged**